apps/web/src/components/templates/BlueprintModal.tsx

```diff
@@ -0,0 +1,114 @@
+import { Modal, useMantineTheme } from '@mantine/core';
+import { colors, shadows, Title, Text, Button } from '../../design-system';
+import { Center, Loader } from '@mantine/core';
+import { useQuery, useMutation } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
+import { createTemplateFromBluePrintId, getBlueprintTemplateById } from '../../api/templates';
+import { When } from '../utils/When';
+import { ActivePageEnum } from '../../pages/templates/editor/TemplateEditorPage';
+import { errorMessage } from '../../utils/notifications';
+import { useEffect, useState } from 'react';
+
+export function BlueprintModal() {
+  const theme = useMantineTheme();
+  const navigate = useNavigate();
+  const onClose = () => {
+    localStorage.removeItem('blueprintId');
+    navigate('/templates', {
+      replace: true,
+    });
+  };
+
+  const [blueprintId, setBluePrintId] = useState<undefined | string>();
+
+  useEffect(() => {
+    const id = localStorage.getItem('blueprintId');
+    setBluePrintId(id === null ? undefined : id);
+  }, [localStorage.getItem('blueprintId')]);
+
+  const { data: blueprint, isInitialLoading: isBluePrintLoading } = useQuery(
+    ['blueprint', blueprintId],
+    () => getBlueprintTemplateById(blueprintId as string),
+    {
+      enabled: !!blueprintId,
+      onError: onClose,
+    }
+  );
+
+  const { mutate, isLoading: isCreating } = useMutation(createTemplateFromBluePrintId, {
+    onSuccess: (template) => {
+      localStorage.removeItem('blueprintId');
+      if (template) {
+        navigate(`/templates/edit/${template?._id}?page=${ActivePageEnum.WORKFLOW}`, {
+          replace: true,
+        });
+      }
+    },
+    onError: (err: any) => {
+      if (err?.message) {
+        errorMessage(err?.message);
+      }
+      onClose();
+    },
+  });
+
+  const isLoading = isBluePrintLoading || isCreating;
+
+  return (
+    <>
+      <Modal
```

Comment:
You could use the Modal in the design system. Seems to have the same design.

Reply:
not really because of the centered title :P